handler/deployment/deployment.go

```diff
@@ -160,17 +160,22 @@ func (d *Deployment) run(ctx context.Context, msg *engine.Message) error {
 	return nil
 }
 
+// TODO This function will be removed once the new queue has been implemented
 func (d *Deployment) deploymentSuccessCheck(ctx context.Context, correlationID, evaluationID, jobID string, jobSpecModifyIndex uint64) error {
 	var jobInfo *api.Job
 	if err := d.get(fmt.Sprintf(infoURL, d.endpoint, jobID), &jobInfo); err != nil {
 		return err
 	}
 	if *jobInfo.Type == api.JobTypeSystem {
-		if err := d.systemDeploymentSuccess(ctx, correlationID, evaluationID, jobID, *jobInfo.Version); err != nil {
+		if err := d.successCheckByAllocations(ctx, correlationID, evaluationID, jobID, *jobInfo.Version); err != nil {
```

Comment:
You could combine these checks using a logical OR operator:

`if *job.Type == api.JobTypeSystem || *job.Type == api.JobTypeBatch { ... }`